src/utils/cli.cpp

```diff
@@ -124,3 +222,49 @@ void ::Cli::printHelp()
         f->printUsage();
     }
 }
+
+void ::Cli::loadConfigFile()
+{
+    std::string conf = Utils::getUserDataLocation() + "/" + CONFIG_FILE;
+
+    // Have we even written one yet?
+    if(Utils::fileExists(conf))
+    {
+        std::string contents = Utils::readFileContents(conf);
+
+        // Strip all comments since the json-parser doesn't support them
+        std::string cleaned = Utils::stripJsonComments(contents);
+
+        if(cleaned.empty())
+            return; // Guess that's not valid
+
+        json j = json::parse(cleaned);
+
+        for (Flag* f : Global::getFlagList())
+            f->readFromConfig(j);
+    }
+}
+
+void ::Cli::writeConfigFile()
+{
+    json j = json::object();
+
+    // Write each flag
+    for (Flag* f : Global::getFlagList())
+        f->writeToConfig(j);
+
+    // Dump using specified indentation. This is important for comments, since they use the same indentation!
+    std::string dumped = j.dump(JSON_DUMP_INDENTATION);
+
+    // Another pass, to insert the comments
+    for (Flag* f : Global::getFlagList())
+        dumped = f->documentConfigText(dumped);
+
+    // Write that one to disk now. Ensure the userdata-folder exists first, though
+    std::string userdata = Utils::getUserDataLocation();
+    Utils::mkdir(userdata);
```

Comment:
Should check for error and return, the current impl will say that it failed to create the file and not the directory which is a bit misleading. Just a nitpick, though.